pkg/assembler/backends/neo4j/cve.go

```diff
@@ -137,3 +143,140 @@ func (e *cveYearToCveID) PropertyNames() []string {
 func (e *cveYearToCveID) IdentifiablePropertyNames() []string {
 	return []string{}
 }
+
+func (c *neo4jClient) Cve(ctx context.Context, cveSpec *model.CVESpec) ([]*model.Cve, error) {
+	session := c.driver.NewSession(neo4j.SessionConfig{AccessMode: neo4j.AccessModeRead})
+	defer session.Close()
+
+	result, err := session.ReadTransaction(
+		func(tx neo4j.Transaction) (interface{}, error) {
+
+			var sb strings.Builder
+			var firstMatch bool = true
+			queryValues := map[string]any{}
+
+			sb.WriteString("MATCH (n:Cve)-[:CveIsYear]->(cveYear:CveYear)-[:CveHasID]->(cveID:CveID)")
+
+			if cveSpec.Year != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "cveYear", "year", "$cveYear")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+					firstMatch = false
+				} else {
+					err := matchAnd(&sb, "cveYear", "year", "$cveYear")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
+				}
+				queryValues["cveYear"] = cveSpec.Year
+			}
+
+			if cveSpec.CveID != nil {
+
+				if firstMatch {
+					err := matchWhere(&sb, "cveID", "id", "$cveID")
+					if err != nil {
+						return nil, fmt.Errorf("string builder failed with err: %w", err)
+					}
```

Comment:
set firstMatch

Reply:
lint error if added as after this point it does not matter if its set to true or false as its last time its used.